tiledb/sm/array_schema/array_schema_evolution.cc

```diff
@@ -156,7 +156,7 @@ Status ArraySchemaEvolution::drop_attribute(const std::string& attribute_name) {
   if (attributes_to_add_map_.find(attribute_name) !=
       attributes_to_add_map_.end()) {
     // Reset the pointer and erase it
-    attributes_to_add_map_[attribute_name].reset(nullptr);
+    attributes_to_add_map_[attribute_name].reset();
     attributes_to_add_map_.erase(attribute_name);
```

Comment:
Can you rely on `erase()` calling the destructor of its entry? If so, you don't need an explicit `reset()` here; in fact, it's better without it.